stl/inc/chrono

```diff
@@ -5465,28 +5465,36 @@ namespace chrono {
         _STL_INTERNAL_CHECK(false);
     }
 
-    template <class _CharT, class _Traits, class _Duration>
-    void _Write_seconds(basic_ostream<_CharT, _Traits>& _Os, const hh_mm_ss<_Duration>& _Val) {
-        _Os << _STD format(_STATICALLY_WIDEN(_CharT, "{:02}"), _Val.seconds().count());
-        if constexpr (hh_mm_ss<_Duration>::fractional_width > 0) {
+    template <unsigned int _Fractional_width, class _CharT, class _Traits, class _Precision>
+    void _Write_seconds(basic_ostream<_CharT, _Traits>& _Os, const seconds& _Seconds, const _Precision& _Subseconds) {
+        _Os << _STD format(_STATICALLY_WIDEN(_CharT, "{:02}"), _Seconds.count());
+        if constexpr (_Fractional_width > 0) {
             _Os << _STD use_facet<numpunct<_CharT>>(_Os.getloc()).decimal_point();
-            if constexpr (treat_as_floating_point_v<typename hh_mm_ss<_Duration>::precision::rep>) {
-                _Os << _STD format(_STATICALLY_WIDEN(_CharT, "{:0{}.0f}"), _STD floor(_Val.subseconds().count()),
-                    _Val.fractional_width);
-            } else {
+            if constexpr (treat_as_floating_point_v<_Precision::rep>) {
                 _Os << _STD format(
-                    _STATICALLY_WIDEN(_CharT, "{:0{}}"), _Val.subseconds().count(), _Val.fractional_width);
+                    _STATICALLY_WIDEN(_CharT, "{:0{}.0f}"), _STD floor(_Subseconds.count()), _Fractional_width);
+            } else {
+                _Os << _STD format(_STATICALLY_WIDEN(_CharT, "{:0{}}"), _Subseconds.count(), _Fractional_width);
             }
         }
     }
 
+    template <class _CharT, class _Traits, class _Duration>
+    void _Write_seconds(basic_ostream<_CharT, _Traits>& _Os, const hh_mm_ss<_Duration>& _Val) {
+        _Write_seconds<_Val.fractional_width>(_Os, _Val.seconds(), _Val.subseconds());

```

Comment:
It's unusual that `_Write_seconds` is overloaded, but the first one requires `_Fractional_width` to be provided as an explicit template argument, while the rest use normal template argument deduction. I believe that these should be named distinctly, because they have distinct usages; I suggest `_Write_fractional_seconds`.